### DISCUSSION TITLE
SplineList Features

### DISCUSSION THREAD

**Discussion by jzwar** [tataratat/splinepy#141] [General]
# Overview
@j042 recently introduced `SplineList` as a way to access splinepy-splines in list form from c++ directly. I would like to provide a way to sample these efficiently, maybe even provide a connectivity for multipatch objects. This is also useful for Multipatch objects.

Questions:
1. What is a good outline for the sample function
    - Do we allow for non-uniform splines, i.e. surfaces and volumes in the same spline-list
    - Do we introduce checks here?
    - Do we sample all splines in one thread and multithread the individual samplings, or do we split up the splines along multiple threads
2. What other functions does Splinelist need?
3. Do we want to provide one mesh for one multipatch if we create a plottable object? / FEM-mesh?

Please let me know what you think

**Comment by j042** [comment]
I started to work on `SplineList` based multithread sampling functions, as part of multi-patch cpp base "project". So, I try to reflect that here:


> 1. What is a good outline for the sample function
>    
>    * Do we allow for non-uniform splines, i.e. surfaces and volumes in the same spline-list

yes, why not? 

>    * Do we introduce checks here?

yes, but perhaps checking parametric-conformity is a bit too much at runtime. As we discussed, we'd like to use some efficient sampling for basis-sharing splines. Perhaps, we can add simple switch (default is False) to indicate that we can use same base.

>    * Do we sample all splines in one thread and multithread the individual samplings, or do we split up the splines along multiple threads

good point. I'd go with the latter, but I guess you are asking, because it can cause unbalanced load, especially if we support any combination of splines. Can think about this

> 2. What other functions does Splinelist need?

Everything.

> 3. Do we want to provide one mesh for one multipatch if we create a plottable object? / FEM-mesh?

Depends, but short answer would be yes.

**Comment by j042** [comment]
#161 removes SplineList and implements PyMultiPatch
